doc/nrf/security/tfm.rst

```diff
@@ -78,128 +93,126 @@ A minimal version of the TF-M secure application is provided in |NCS| to show ho
+Code and memory size of TF-M increases as more services are enabled but also selected hardware places limitations on how separation of secure and non-secure are made.
+
+TF-M is linked as a separate partition in the final binary image. The reserved sizes of its RAM and Flash partitions are configured by the :kconfig:option:`CONFIG_PM_PARTITION_SIZE_TFM` and :kconfig:option:`CONFIG_PM_PARTITION_SIZE_TFM_SRAM` options.
+These configuration option allows you to specify the size allocated for the TF-M partition in the final binary image. Default partition sizes vary between device families and are not optimized to any specific use case.
+
+Process to optimize the TF-M size is to first find the minimal set of features to satisfy the application needs and then minimize the allocated partition sizes while still conforming to the alignment and granularity requirements of given hardware.
+See next section for more information about the alignment requirements.
 
 .. _ug_tfm_partition_alignment_requirements:
 
 TF-M partition alignment requirements
-*************************************
+=====================================
 
 TF-M requires that secure and non-secure partition addresses must be aligned to the flash region size :kconfig:option:`CONFIG_NRF_TRUSTZONE_FLASH_REGION_SIZE`.
 |NCS| ensures that they in fact are aligned and comply with the TF-M requirements.
 
 On nRF53 and nRF91 Series devices, TF-M uses the SPU to enforce the security policy between the partitions, so the :kconfig:option:`CONFIG_NRF_TRUSTZONE_FLASH_REGION_SIZE` is set to the SPU flash region size.
 On nRF54L15 devices, TF-M uses the MPC to enforce the security policy between the partitions, so the :kconfig:option:`CONFIG_NRF_TRUSTZONE_FLASH_REGION_SIZE` is set to the MPC region size.
 
+.. list-table:: Region limits on different hardware
+   :header-rows: 1
+
+   * - Family
+     - RAM granularity
+     - ROM granularity
+   * - nRF91 Series
+     - 8 kB
+     - 32 kB
+   * - nRF53 Series
+     - 8 kB
+     - 16 kB
+   * - nRF54 Series
+     - 4 kB
+     - 4 kB
+
+.. figure:: /images/nrf-secure-rom-granularity.svg
+   :alt: Partition alignment granularity
+   :width: 80%
+   :align: center
+
+   Partition alignment granularity on different nRF devices.
+
 When the :ref:`partition_manager` is enabled, it will take into consideration the alignment requirements.
 But when the static partitions are used, the user is responsible for following the alignment requirements.
 
+.. figure:: /images/secure-flash-regions.svg
+   :alt: Example of aligning partitions with flash regions
+   :width: 80%
+   :align: center
+
+   Example of aligning partitions with flash regions.
+
 If you are experiencing any partition alignment issues when using the Partition Manager, check the :ref:`known_issues` page on the main branch.
 
 The partitions which need to be aligned with the TrustZone flash region size are partitions ``tfm_nonsecure`` and ``nonsecure_storage``.
```

Comment:
tfm_storage also needs to be aligned with this, if it is set after non-secure partition.